### DISCUSSION TITLE
Collaborative labeling of fly videos

### DISCUSSION THREAD

**Discussion by sahilsingh2402** [talmolab/sleap#1336] [Help!] [answered]
Hello! I've been working on labeling fly videos that feature 9 to 10 flies each. It takes me approximately an hour to label one frame with all the flies in it. I was wondering if there's a way for me to share this task with my lab mates. Specifically, can we divide the work such that I label the first 50 frames while someone else labels the next 50 frames? Our video tends to be pixelated and noisy due to the small size of the flies in relation to the arena, which makes the labeling process quite labor-intensive. Thank you for any help you can provide!

**Comment by talmo** [comment, marked as answer]
Hi @sahilsingh2402,

We do not currently have plans to support collaborative labeling on the same project, but there are a couple of strategies you can use that should mitigate this issue:

1. Split up the videos across multiple projects for concurrent annotation and merge them later. If there's a single video, you can break it up into chunks using ffmpeg.
2. Train the model for prediction using only 1-2 frames. Especially for top-down which works well for fly data, you can get away with training on few frames if you have many animals within each since each one is treated as a separate sample. After training, the predictions imported will be easier to correct than labeling from scratch.

Let us know if you have any questions!

Talmo

**Reply by sahilsingh2402** [reply]
Thank you for the helpful suggestions :)

**Reply by roomrys** [reply]
Hi @sahilsingh2402,

We had another user who had many small animals in a single frame ([see this thread](https://github.com/talmolab/sleap/discussions/1035#discussioncomment-6063595)). They split their labeling by using cropped images of the animals as training input - thus, they only had to finish labeling one crop/animal for it to be a valid training input. This is not inherently part of the SLEAP pipeline, but it seemed worth mentioning since you have a similar problem. Perhaps you could look into doing something similar?

The top-down model in SLEAP first uses a centroid model to find the center of the animal and then output a centered crop. If you could generate a directory of cropped images output from a centroid cropping model (either by saving the output of SLEAP's centroid model, assuming you have good centroid predictions already, or using another program to output these crops), then you can use those images as the training input for your model. SLEAP supports loading a directory of images as a "video" to be annotated. I am not sure how @WeheliyeHashi retrieved their cropped images, but perhaps they can chime in?

Thanks,
Liezl

**Reply by WeheliyeHashi** [reply]
Hi @roomrys ,

Our videos were quite simple to label.  As you know, the resolutions of our videos are 3036 by 3036 and our worms are of size 100 pixels. This meant that we needed to localize the worm center and draw a bounding box that captures the minimum and maximum coordinates of each worm.

To do this, we followed the following steps:

We used adaptive thresholding to remove the background and keep the foreground (the worms).
We found the center of each worm and placed a bounding box around it.
For two or more worms that were attached, we applied step 2 and found the center of the combined worms.
For each bounding box, we manually labeled the skeleton of the worm.
We added padding to each label to make them the same size (around 512 by 512 pixels).
We ran the labels through the Sleapai GUI.

Hope that helps !

**Reply by sahilsingh2402** [reply]
Thank you all for your help, as I was able to accurately predict the presence of 10 flies with 90-95% accuracy despite having noisy data. However, it took at least 15 labeled frames for the network to learn and predict correctly. To accomplish this, I utilized a top-down model approach with Unet for the centroid model and a pre-trained encoder for the centroid instance model. Overall, the results were very successful, and I appreciate @talmo 's suggestion.
